dinky-admin/src/main/java/org/dinky/service/impl/TaskServiceImpl.java

```diff
@@ -1048,9 +1051,12 @@ public JobInstance refreshJobInstance(Integer id, boolean isCoercive) {
             History history = historyService.getById(jobInstance.getHistoryId());
             history.setConfig(JSONUtil.parseObject(history.getConfigJson()));
             if (Asserts.isNotNull(history.getClusterConfigurationId())) {
-                jobInfoDetail.setClusterConfiguration(
+                ClusterConfiguration clusterConfigById =
                         clusterConfigurationService.getClusterConfigById(
-                                history.getClusterConfigurationId()));
+                                history.getClusterConfigurationId());
+                jobInfoDetail.setClusterConfiguration(clusterConfigById);
+                Task task = getById(jobInstance.getTaskId());
+                jobInfoDetail.getInstance().setType(task.getType());
```

Comment:
Why set type to instance?

Reply:
in handleJobDone function
at line 1428
`jobConfig.getGatewayConfig().setType(GatewayType.get(jobInstance.getType()));`
buildGatewayConfig need task type
we need to delete flink container when job has finished

Reply:
> Why set type to instance?

in handleJobDone function
at line 1428
`jobConfig.getGatewayConfig().setType(GatewayType.get(jobInstance.getType()));`
buildGatewayConfig need task type
we need to delete flink container when job has finished

Reply:
history.getType()